src/merkle_tree.rs

```diff
@@ -37,18 +41,22 @@ impl MerkleTree for MantaCrypto {
 	type Tree = LedgerMerkleTree;
 
 	/// build a merkle tree from the leaves
-	fn build_tree(hash_param: Self::Param, leaves: &[Self::Leaf]) -> Self::Tree {
-		LedgerMerkleTree::new(hash_param, leaves).unwrap()
+	fn build_tree(
+		hash_param: Self::Param,
+		leaves: &[Self::Leaf],
+	) -> Result<Self::Tree, MantaErrors> {
+		let tree = LedgerMerkleTree::new(hash_param, leaves)?;
```

Comment:
is there any reason to use `?` and then `Ok()`? I think this can just be
```suggestion
		LedgerMerkelTree::new(hash_param, leaves)
```

Reply:
Sounds good to me.